src/Form/tests/FormControl.test.jsx

```diff
@@ -1,26 +1,63 @@
-import React from 'react';
-import { mount } from 'enzyme';
+import React, { useState } from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
 
 import FormControl from '../FormControl';
 
 const ref = {
   current: null,
 };
 
+// eslint-disable-next-line react/prop-types
+function Component({ isClearValue }) {
+  const onChangeFunc = jest.fn();
+  const [inputValue, setInputValue] = useState('');
+
+  return (
+    <FormControl
+      hasInputMask
+      mask="+{1}(000)000-00-00"
+      value={inputValue}
+      onChange={isClearValue ? onChangeFunc : (e) => setInputValue(e.target.value)}
+      onAccept={isClearValue ? onChangeFunc : (value) => setInputValue(value)}
+      data-testid="form-control-with-mask"
+    />
+  );
+}
+
 describe('FormControl', () => {
   it('textarea changes its height with autoResize prop', () => {
     const useReferenceSpy = jest.spyOn(React, 'useRef').mockReturnValue(ref);
     const onChangeFunc = jest.fn();
-    const wrapper = mount((
-      <FormControl as="textarea" autoResize onChange={onChangeFunc} />
-    ));
+    render(
+      <FormControl as="textarea" autoResize onChange={onChangeFunc} data-testid="form-control-textarea" />,
+    );
+
     ref.scrollHeight = 180;
     ref.offsetHeight = 90;
     ref.clientHeight = 88;
+
+    const textarea = screen.getByTestId('form-control-textarea');
+
     expect(useReferenceSpy).toHaveBeenCalledTimes(1);
     expect(ref.current.style.height).toBe('0px');
-    wrapper.find('textarea').simulate('change');
+
+    fireEvent.change(textarea, { target: { value: 'new text' } });
+
     expect(onChangeFunc).toHaveBeenCalledTimes(1);
-    expect(ref.current.style.height).toEqual(`${ref.current.scrollHeight + ref.current.offsets}px`);
+    expect(ref.current.style.height).toEqual(`${ref.current.scrollHeight + ref.current.offsetHeight}px`);
+  });
+  it('should apply and accept input mask for phone numbers', () => {
+    render(<Component />);
+
+    const input = screen.getByTestId('form-control-with-mask');
+    fireEvent.change(input, { target: { value: '1234567890' } });
```

Comment:
Please, use `userEvent` instead

Reply:
This is not possible for the pure value example. To pass a value and receive a cleared one, the [react-imask](https://github.com/uNmAnNeR/imaskjs) library provides an `onAccept` property, which does not work when using `userEvent`. Inside `react-imask`, processing of the value passed to `onAccept` occurs through manual transmission and processing (by library tools), in contrast to the familiar input to the `onChange` property.
Let's leave `fireEvent` here since this is the only tool that can recognize the cleared value at the moment.